main.c

```diff
@@ -371,6 +371,10 @@ int main(int argc, char *argv[]) {
 		filepath(output_filepath, output_filename);
 	} else {
 		output_filename = argv[optind];
+		if (strlen(output_filename) >= PATH_MAX) {
```

Comment:
Should be `PATH_MAX - 1` because we want a terminating NULL char

Reply:
strlen() gets the length of the string excluding the NULL char, and since it's a >= comparison it will exit if there is no room for the NULL char.

Reply:
`strlen` excludes the terminating NULL char, `PATH_MAX` includes it

Reply:
Bleh, math is hard